### PR TITLE
Add catalog name and version to the subscription

### PR DESCRIPTION
**Pull request by Skarlso**
### Description

https://github.com/weaveworks/profiles/issues/87

This is the first part of this issue where we add in these values to the subscription. [pctl](https://github.com/weaveworks/pctl/) will provide the rest of the functionality by filling these values in from the catalog service.

**EDIT**: This will now also change the reconciliation process to fetch profileurl and branch based on catalogname, profilename and version.

### Checklist
- [x] Added tests that cover your change
- [x] Added/modified documentation as required (such as the `README.md` (diagrams, usage, roadmap etc), and anything in `examples/`)
- [x] Made sure the title of the PR is a good description that can go into the release notes
- [x] Added a `kind` label to the PR (e.g. `kind/feature`)

### CODE REVIEW

**Review by aclevername — COMMENTED** [review]
can we update the acceptance test to use this new method?

File: api/v1alpha1/profilesubscription_types.go [hunk 1]
```diff
@@ -49,6 +51,26 @@ type ProfileSubscriptionSpec struct {
...
+// ProfileCatalogDescription defines properties of the catalog this profile is from
+type ProfileCatalogDescription struct {
+	// Version defines the version of the catalog to get the profile from
+	Version string `json:"version,omitempty"`
+
+	// Catalog defines the name of the catalog to get the profile from
+	Catalog string `json:"catalog,omitempty"`
+
+	// ProfileName defines the name of the profile
+	ProfileName string `json:"profile_name,omitempty"`
```
Comment (aclevername) [inline api/v1alpha1/profilesubscription_types.go:68]: Since we don't say `CatalogName`, could we rename `ProfileName` to `Profile`? What do you think?

Comment (Skarlso) [reply api/v1alpha1/profilesubscription_types.go:68]: Sure, that makes sense.

**Review by aclevername — APPROVED** [review]
LGTM once renaming `ProfileName` :smile:

**Pull request merged**